src/coreComponents/constitutive/fluid/PVTFunctions/FenghourCO2ViscosityFunction.cpp

```diff
@@ -13,158 +13,152 @@
-
-  value.m_var = viscosity.m_var;
-  value.m_der[0] = viscosity.m_der[0];
-}
-
-void FenghourCO2ViscosityFunction::fenghourCO2Viscosity( real64 const & Tcent, real64 const & den, real64 & vis )
+void FenghourCO2Viscosity::fenghourCO2ViscosityFunction( real64 const & temperatureCent,
+                                                         real64 const & density,
+                                                         real64 & viscosity )
 {
   constexpr real64 espar = 251.196;
   constexpr real64 esparInv = 1.0 / espar;
+  // coefficients from Table (1) of Fenghour and Wakeham (1998)
   constexpr real64 aa[5] = { 0.235156, -0.491266, 5.211155e-2, 5.347906e-2, -1.537102e-2 };
+  // coefficients from Table (3) of Fenghour and Wakeham (1998)
   constexpr real64 d11 =  0.4071119e-2;
   constexpr real64 d21 =  0.7198037e-4;
   constexpr real64 d64 =  0.2411697e-16;
   constexpr real64 d81 =  0.2971072e-22;
   constexpr real64 d82 = -0.1627888e-22;
-
-  // temperature in Kelvin
-  const real64 Tkelvin = Tcent + 273.15;
-  // evaluate vlimit from eqns 3-5
-  const real64 Tred   = Tkelvin * esparInv;
-  const real64 x = log( Tred );
-  const real64 lnGfun = aa[0] + x * (aa[1] + x * (aa[2] + x *(aa[3] + x * aa[4])));
-  const real64 GfunInv = exp( -lnGfun );
-  const real64 vlimit = 1.00697 * sqrt( Tkelvin ) * GfunInv;
-
-  const real64 d2 = den * den;
-  const real64 vxcess = den * (d11 + den * (d21 + d2*d2*(d64 / (Tred*Tred*Tred) + d2*(d81 + d82/Tred))));
-
+  // we neglect critical viscosity
   constexpr real64 vcrit = 0.0;
 
-  vis = 1e-6 * (vlimit + vxcess + vcrit);
+  // temperature in Kelvin
+  real64 const temperatureKelvin = temperatureCent + 273.15;
+  // equation (5) of Fenghour and Wakeham (1998)
+  real64 const Tred = temperatureKelvin * esparInv;
+  real64 const x = log( Tred );
+  // equation (4) of Fenghour and Wakeham (1998)
+  real64 const lnGfun = aa[0] + x * (aa[1] + x * (aa[2] + x *(aa[3] + x * aa[4])));
+  real64 const GfunInv = exp( -lnGfun );
+  // equation (3) of Fenghour and Wakeham (1998)
+  real64 const vlimit = 1.00697 * sqrt( temperatureKelvin ) * GfunInv;
+
+  // equation (8) of Fenghour and Wakeham (1998)
+  real64 const d2 = density * density;
+  real64 const vxcess = density * (d11 + density * (d21 + d2*d2*(d64 / (Tred*Tred*Tred) + d2*(d81 + d82/Tred))));
+
+  // equation (1) of Fenghour and Wakeham (1998)
+  viscosity = 1e-6 * (vlimit + vxcess + vcrit);
 }
 
-void FenghourCO2ViscosityFunction::calculateCO2Viscosity( real64_array const & pressure, real64_array const & temperature, real64_array2d const & density,
-                                                          real64_array2d const & viscosity )
+void FenghourCO2Viscosity::calculateCO2Viscosity( array1d< array1d< real64 > > const & coordinates,
+                                                  array1d< real64 > const & valuesDensity,
+                                                  array1d< real64 > const & valuesViscosity )
```

Comment:
How is `valuesDensity` and `valuesViscosity` differ from `densities` and `viscosities`.
I like the simple names most, but it's a matter of taste.

Reply:
I changed the names to `densities` and `viscosities`